yfinance/utils.py

```diff
@@ -470,45 +474,47 @@ def fix_Yahoo_returning_live_separate(quotes, interval, tz_exchange):
     # Fix = merge them together
     n = quotes.shape[0]
     if n > 1:
-        dt1 = quotes.index[n-1]
-        dt2 = quotes.index[n-2]
+        dt1 = quotes.index[n - 1]
+        dt2 = quotes.index[n - 2]
         if quotes.index.tz is None:
             dt1 = dt1.tz_localize("UTC")
             dt2 = dt2.tz_localize("UTC")
         dt1 = dt1.tz_convert(tz_exchange)
         dt2 = dt2.tz_convert(tz_exchange)
-        if interval in ["1wk", "1mo", "3mo"]:
+        if interval == "1d":
+            # Similar bug in daily data except most data is simply duplicated
+            # - exception is volume, *slightly* greater on final row (and matches website)
+            if dt1.date() == dt2.date():
+                # Last two rows are on same day. Drop second-to-last row
+                quotes = quotes.drop(quotes.index[n - 2])
+        else:
             if interval == "1wk":
-                last_rows_same_interval = dt1.year==dt2.year and dt1.week==dt2.week
+                last_rows_same_interval = dt1.year == dt2.year and dt1.week == dt2.week
             elif interval == "1mo":
-                last_rows_same_interval = dt1.month==dt2.month
+                last_rows_same_interval = dt1.month == dt2.month
             elif interval == "3mo":
-                last_rows_same_interval = dt1.year==dt2.year and dt1.quarter==dt2.quarter
-            if last_rows_same_interval:
+                last_rows_same_interval = dt1.year == dt2.year and dt1.quarter == dt2.quarter
+            else:
+                last_rows_same_interval = None
+
+            if last_rows_same_interval is not None:
```

Comment:
This change has modified behaviour, now it's always dropping last row even when it shouldn't. `last_rows_same_interval` default must be False not None, and the `if` must check for True/False

If you aren't convinced by this then do the following:

- in base.py, duplicate the call to `utils.fix_Yahoo_returning_live_separate()` so it is called twice

Then fetch weekly data. The first call will work correctly, fixing a Yahoo error. But the second call, acting on data which no longer has the Yahoo error, will remove all of this weeks data - incorrect.

=> So if the data never had the Yahoo error, e.g. because day was Monday, then first call will remove this weeks data.

Reply:
Good catch, have updated the PR